<launch>
  <arg name="launch_prefix" default="" /> <!-- set to value="gdbserver localhost:10000" for remote debugging -->
  <arg name="node_namespace" default="apriltag_ros_continuous_node" />
  <arg name="camera_name" default="/zed2/zed_node/left" />
  <!-- <arg name="image_topic" default="image/uncompressed" /> -->

  <!-- Set parameters -->
  <rosparam command="load" file="$(find jackal_ros)/config/settings.yaml" ns="$(arg node_namespace)" />
  <rosparam command="load" file="$(find jackal_ros)/config/tags.yaml" ns="$(arg node_namespace)" />

 <param name="/use_sim_time" value="false" type="bool" />

  <!-- Uncompress images and convert into greyscale -->
  <node pkg="image_transport" type="republish" name="image_republisher0" required="true" args="compressed in:=/zed2/zed_node/left/image_rect_color raw out:=$(arg camera_name)/uncompressed _image_transport:=compressed" />
  <!-- <node pkg="image_transport" type="republish" name="image_republisher1" required="true" args="compressed in:=/zed2/zed_node/right/image_rect_color raw out:=/image1/uncompressed _image_transport:=compressed" /> -->

  <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="$(arg node_namespace)" clear_params="true" output="screen" required="true" >
    <!-- Remap topics from those used in code to those on the ROS network -->
    <remap from="image_rect" to="$(arg camera_name)/uncompressed" />
    <!-- <remap from="camera_info" to="$(arg camera_name)/camera_info" /> -->

    <param name="publish_tag_detections_image" type="bool" value="true" />      <!-- default: false -->
  </node>
</launch>